nexus/tests/integration_tests/disks.rs

```diff
@@ -852,15 +770,18 @@ async fn test_disk_region_creation_failure(
             test.sled_agent.get_crucible_dataset(test.zpool_id, *id).await;
         crucible.set_create_callback(Box::new(|_| RegionState::Created)).await;
     }
-    let _: Disk = objects_post(&client, &disks_url, new_disk.clone()).await;
+    let _ = create_disk(client, ORG_NAME, PROJECT_NAME, DISK_NAME).await;
 }
 
 async fn disk_get(client: &ClientTestContext, disk_url: &str) -> Disk {
     object_get::<Disk>(client, disk_url).await
 }
 
 async fn disks_list(client: &ClientTestContext, list_url: &str) -> Vec<Disk> {
-    objects_list_page::<Disk>(client, list_url).await.items
+    NexusRequest::iter_collection_authn(client, list_url, "", 10)
```

Comment:
nit: 10 is a magic number, could we make this a constant?

Reply:
Sure.  There are 5-10 other callers, and nearly all of them don't really care what value is provided here.  So I've changed `iter_collection_authn` to accept an `Option<usize>` and put the named constant inside that function.  It's used if you pass `None`.